feature/text_based.py

```diff
@@ -0,0 +1,703 @@
+        is_row_covered = [model.add_var(var_type=BINARY) for _ in data.rows]
+        num_row_covered = model.add_var(var_type=INTEGER)
+
+        # Constraint: Link between x and is_row_covered
+        for row in data.rows:
+            for i in data.X:
+                # if any selected column has the label, then the row would be covered
+                model.add_constr(data.matrix[row, i] * x[i] <= is_row_covered[row])
+            # total selected
+            model.add_constr(xsum(data.matrix[row, i] * x[i] for i in data.X) >= is_row_covered[row])
+
+        # Constraint: Link is_row_covered with num_row_covered
+        model.add_constr(xsum(is_row_covered[row] for row in data.rows) == num_row_covered)
+
+        # Constraint: If selected is given, discard columns that are not part of selection
+        for i in data.X:
+            if i not in selected:
+                model.add_constr(x[i] == 0)
+
+        # Constraint: limit number of selected to max_cover_size
+        model.add_constr(xsum(x[i] for i in data.X) <= self.selection_size)
+
+        # Objective: maximize "row" coverage (not the whole coverage of 1s)
+        model.objective = maximize(xsum(is_row_covered[row] for row in data.rows))
+
+        # Solve
+        model.verbose = False
+        model.optimize()
+
+        # Solution
+        selected = [i for i in data.X if float(x[i].x) >= 0.99]
+
+        if self.verbose:
+            print("=" * 40)
+            print("MAX COVER OBJECTIVE:", model.objective_value)
+            print("NUM ROWS COVERED:", num_row_covered.x,
+                  "coverage: {:.2f}".format(num_row_covered.x / data.matrix.shape[0]))
+            print("SIZE:", len(selected),
+                  "reduction: {:.2f}".format((data.matrix.shape[1] - len(selected)) / data.matrix.shape[1]))
+            print("SELECTED:", selected)
+            print("STATUS:", model.status)
+            print("=" * 40)
+
+        assert model.status == OptimizationStatus.OPTIMAL
+
+        # Return
+        return selected
+
+    @staticmethod
+    def _validate_args(selection_size):
+        if selection_size is not None:
+            if selection_size <= 0:
+                raise ValueError("Selection size must be greater than zero.")
+
+
+# TODO not sure if this should use _BaseDispatcher or not.
+class _TextBased(_BaseSupervisedSelector):
+    """
+    The fit() method is responsible for featurization and optimization.
+    The transform() method is responsible for selecting features/content.
+    """
+
+    def __init__(self, seed: int,
+                 num_features: Num,
+                 featurization_method: TextWiser,
+                 optimization_method: str,
+                 cost_metric: str,
+                 trials: int = 10):
+
+        # Call constructor of parent class _BaseSupervisedSelector
+        super().__init__(seed)
+
+        # Set input arguments
+        self.num_features = num_features
```

Comment:
why do we have both self.num_features and self.selection_size? Cannot remove one (say selection_size)

Reply:
Yes, we can. Removed selection_size and replaced with num_features